Adding multi line highlighting

aj2duncan opened the pull request:
As discussed in #53 added multiline highlighting. Tested using
````
---
title: "My Problem"
output:
  xaringan::moon_reader:
    lib_dir: libs
    nature:
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
---

```{r}
add1 <- function(x) x + 1
```

---

```{r}
## works
{{add1(2)}}
```

---

```{r}
## works
{{add1(
  add1(2)
)}}
```

---

```{r}
## works
add1(
  {{add1(2)}}
)
```

---

```{r}
## works
my_fn = function(){
{{data.frame(x = rnorm(10),
          y = rnorm(10))}}
}
```

---

```{r}
## works
my_fn = function(){
  {{num = 10}}
  
{{  dat = data.frame(x = rnorm(num),
                   y = rnorm(num))}}
  
{{  dat = cbind(dat, 
              z = 2*dat$x)}}
}
```
````
which seems to work. The only thing it doesn't sort automatically is indentation as you can see from the last two examples. Rmd, libs and html attached in zip file.
[xaringan_test.zip](https://github.com/yihui/xaringan/files/1075592/xaringan_test.zip)

yihui commented:
Much appreciated! Before I find time to review it, would you mind moving the code you added to a separate function in utils.R? That way, the body of `moon_reader` will be smaller.

yihui commented on xaringan.Rproj:
```diff
@@ -3,6 +3,7 @@ Version: 1.0
 RestoreWorkspace: Default
 SaveWorkspace: Default
 AlwaysSaveHistory: Default
+QuitChildProcessesOnExit: Default
```
Which version of RStudio are you using? I have never seen such an option before. Usually I prefer not adding irrelevant changes to a PR if possible.

aj2duncan commented:
Sorry about the erroneous line in `xaringan.Rproj` - it is from the RStudio daily builds. I've moved the highlighting to a separate function and removed that line. 

Thanks.

yihui commented:
Thanks for your efforts, and sorry for the delay. I just managed to breathe a little bit after the "blogdown marathon" that lasted several months.

I looked at your implementation for a while and had a hard time understanding it, so I went ahead and implemented it by myself, which is probably simpler than your PR. Anyway, your contribution is very much appreciated!

aj2duncan commented:
Thanks for fixing this @yihui and your method it certainly simpler than mine!
